samples/powerpoint/custom-functions/search-wikipedia-api.yaml

```diff
@@ -0,0 +1,221 @@
+                var key = Object.keys(data.query.pages)[0];
+                if (data.query.pages[key].title !== "Undefined") {
+                  var img = data.query.pages[key].thumbnail.source;
+                  $(itemHTML)
+                    .find(".ms-ListItem-image")
+                    .append('<img width="70px" src="' + img + '" />');
+                }
+              }
+            });
+        }
+
+        /* Wikipedia API query */
+        function wiki_base(pattern) {
+          const base_url = "https://en.wikipedia.org/w/api.php";
+          const base_query = "?origin=*&action=query&format=json&prop=";
+          return base_url + base_query + pattern;
+        }
+
+        function build_wiki_search_url(pattern) {
+          let qry = "pageimages&list=search&srsearch=";
+          return wiki_base(qry + pattern);
+        }
+
+        function build_wiki_image_search(pattern) {
+          let qry = "pageimages&piprop=original&pilicense=any&pageids=";
+          return wiki_base(qry + pattern);
+        }
+
+        function build_wiki_file_search(pattern) {
+          let qry = "pageimages|pageterms&pilicense=any&titles=";
+          return wiki_base(qry + pattern);
+        }
+
+        function build_wiki_pageinfo(pattern) {
+          let qry = "info|images&inprop=url&pageids=";
+          return wiki_base(qry + pattern);
+        }
+
+        /* Render */
+        function build_list_item(title, pageid, summary, ts) {
+          return (
+            '<li class="ms-ListItem ms-ListItem--image" tabindex="0" data-pageid="' +
+            pageid +
+            '">' +
+            '<div class="ms-ListItem-image" ></div>' +
+            '<span class="ms-ListItem-secondaryText listItem-link">' +
+            title +
+            "</span>" +
+            "<span>" +
+            summary +
+            "</span>" +
+            '<div class="ms-ListItem-selectionTarget" > </div>' +
+            '<div class="ms-ListItem-actions" >' +
+            '<div class="ms-ListItem-action" >' +
+            '<i class="ms-Icon ms-Icon--Flag" > </i>' +
+            "</div>" +
+            '<div class="ms-ListItem-action" >' +
+            '<i class="ms-Icon ms-Icon--Pinned" > </i>' +
+            "</div>" +
+            "</div>" +
+            "</li>"
+          );
+        }
+    language: typescript
+template:
+    content: |-
+        <header class="ms-welcome__header ms-bgColor-neutralLighter" style="text-align: center">
+          <h1 class="ms-font-su">Search Wikipedia</h1>
+        </header>
+
+        <section class="ms-font-m">
+          <p>This sample shows how to query external API using selected text in the presentation.</p>
+          <p>The result then will be displayed in the task pane.</p>
+        </section>
+
+        <section class="setup ms-font-m">
+          <h3>Try it out</h3>
+          <p>The search result will be provided by Wikipedia API. Select on any word/sentence in the presentation and, click "Search"</p>
```

Comment:
Consider changing this line to
"Select any text in the presentation and press <b>Search</b> to see related Wikipedia entries."